### PR TITLE
libarchive: Support Conan V2

### PR DESCRIPTION
**Pull request by jwillikers**
Specify library name and version:  **libarchive/***

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libarchive/all/conanfile.py [hunk 1]
```diff
@@ -116,110 +116,118 @@ def requirements(self):
...
-                              "LIST(APPEND CMAKE_MODULE_PATH")
+        replace_in_file(self, cmakelists_path,
+                        "SET(CMAKE_MODULE_PATH",
+                        "LIST(APPEND CMAKE_MODULE_PATH")
         # allow openssl on macOS
         if self.options.with_openssl:
-            tools.replace_in_file(cmakelists_path,
-                                  "IF(ENABLE_OPENSSL AND NOT CMAKE_SYSTEM_NAME MATCHES \"Darwin\")",
-                                  "IF(ENABLE_OPENSSL)")
+            replace_in_file(self, cmakelists_path,
```
Comment (prince-chrismc) [inline recipes/libarchive/all/conanfile.py:187]: I feel like these could be patches bit of have no clue if that's reasonable

**Review by prince-chrismc — APPROVED** [review]

File: recipes/libarchive/all/conanfile.py (hunk 1 above)
Comment (jwillikers) [reply recipes/libarchive/all/conanfile.py:187]: I can look into that once I get the pipeline passing

Comment (jwillikers) [reply recipes/libarchive/all/conanfile.py]: That was painful, but done.

File: recipes/libarchive/all/conandata.yml [hunk 2]
```diff
@@ -19,8 +19,52 @@ sources:
...
 patches:
   "3.6.1":
-    - patch_file: "patches/android_compile.patch"
+    - patch_file: "patches/0001-3.6.0-zlib-winapi.patch"
+      patch_description: "Remove broken ZLIB WINAPI check"
+      patch_type: "portability"
+    - patch_file: "patches/0003-3.6.0-cmake.patch"
+      patch_description: "Make CMake build-system compatible with Conan"
+      patch_type: "conan"
+    - patch_file: "patches/0004-3.6.0-android.patch"
```
Comment (prince-chrismc) [inline recipes/libarchive/all/conandata.yml:28]:
```suggestion
    - patch_file: "patches/0004-3.6.0-android.patch"
       patch_description: "????"
```

**Review by prince-chrismc — DISMISSED** [review]
Thank you sooo much, its a much better to review ❤️

File: recipes/libarchive/all/conandata.yml (hunk 2 above)
Comment (jwillikers) [reply recipes/libarchive/all/conandata.yml:28]: Fixed.

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

File: recipes/libarchive/all/conanfile.py [hunk 3]
```diff
@@ -116,110 +116,68 @@ def requirements(self):
...
         # too strict check
-        self._cmake.definitions["ENABLE_WERROR"] = False
-        if tools.Version(self.version) >= "3.4.2":
-            self._cmake.definitions["ENABLE_MBEDTLS"] = self.options.with_mbedtls
-        self._cmake.definitions["ENABLE_XATTR"] = self.options.with_xattr
-
-        self._cmake.configure(build_folder=self._build_subfolder)
-        return self._cmake
-
-    def _patch_sources(self):
```
Comment (SpaceIm) [inline recipes/libarchive/all/conanfile.py:168]: not sure it was a good idea to move these replacement to patches, it's more work for every new version with the risk to have a bad patch.

File: recipes/libarchive/all/CMakeLists.txt [hunk 4]
```diff
@@ -1,14 +0,0 @@
...
-
-include(conanbuildinfo.cmake)
-conan_basic_setup(KEEP_RPATHS)
-
-if(CMAKE_CROSSCOMPILING)
-    if(WIN32 AND NOT CYGWIN)
-        set(ZLIB_WINAPI_EXITCODE "0")
-        set(ZLIB_WINAPI_EXITCODE__TRYRUN_OUTPUT "")
-    endif()
-endif()
```
Comment (SpaceIm) [inline recipes/libarchive/all/CMakeLists.txt:12]: what happened to this thing?

**Pull request merged**